build_tools/packaging/promote_from_rc_to_final.py

```diff
@@ -0,0 +1,278 @@
+    # but inside we have to match to rocm_sdk_libraries_gfx94X-dcgpu/ with a capital "X-dcgpu" instead of "x_dcgpu"
+    for dir in glob.glob(str(new_dir_path) + "/*"):
+        stripped_dir = str(dir).rsplit("/", 1)[-1]
+        if (
+            "gfx" in package_name_no_version
+            and "gfx" in stripped_dir
+            and len(package_name_no_version) == len(stripped_dir)
+        ):
+            package_name_no_version = stripped_dir
+            break
+
+    old_rocm_version = (
+        str(old_version)
+        if not "rocm" in str(old_version)
+        else str(old_version).split("+rocm")[-1]
+    )
+    new_rocm_version = (
+        str(new_version)
+        if not "rocm" in str(new_version)
+        else str(new_version).split("+rocm")[-1]
+    )
+
+    print("  Changing wheel-specific files that contain the version", end="")
+
+    if not "torch" in str(new_dir_path):  # rocm packages
+        files_to_change = [
+            f"{new_dir_path}/{package_name_no_version}/_dist_info.py",
+        ]
+    elif (
+        "torch" == package_name_no_version
+    ):  # only exactly torch and not triton, torchaudio, ..
+        print("torch")
+        files_to_change = [
+            f"{new_dir_path}/{package_name_no_version}/_rocm_init.py",
+            f"{new_dir_path}/{package_name_no_version}/version.py",
+        ]
+
+        # special handling
+        # we only want to change required-distr matching "rocm"
+        with fileinput.input(
+            files=(
+                f"{new_dir_path}/{package_name_no_version}-{old_version}.dist-info/METADATA"
+            ),
+            encoding="utf-8",
+            inplace=True,
+        ) as f:
+            for line in f:
+                if "Requires-Dist" in line:
+                    if "rocm" in line:
+                        print(line.replace(old_rocm_version, new_rocm_version), end="")
+                        continue
+                print(line, end="")
+    elif not "triton" in package_name_no_version:  # torchaudio, torchvision
+        files_to_change = [
+            f"{new_dir_path}/{package_name_no_version}/version.py",
+        ]
+    else:  # triton
+        return
+
+    with fileinput.input(files=(files_to_change), encoding="utf-8", inplace=True) as f:
+        for line in f:
+            print(line.replace(old_rocm_version, new_rocm_version), end="")
+
+    print(" ...done")
+
+
+def promote_wheel(filename):
+    print(f"Promoting whl from rc to final: {filename}")
+
+    original_wheel = Wheel(filename)
+    original_version = Version(original_wheel.version)
+    base_version = original_version.base_version
+
+    print(f"  Detected version: {original_version}")
+
+    if original_version.local:  # torch packages
+        if not "rc" in original_version.local:
+            print("  Only release candidates (rc) can be promoted! Aborting!")
+            return False
+        local_version = str(original_version.local).split("rc", 1)[0]
```

Comment:
The variable naming here should be improved. I was a bit confused what this should do at a first glance, maybe call it "new_local_version" or "promoted_local_version"?